modules/OCPP201/OCPP201.cpp

```diff
@@ -319,34 +398,35 @@ void OCPP201::init() {
             case types::evse_manager::SessionEventEnum::TransactionStarted: {
                 const auto transaction_started = session_event.transaction_started.value();
                 const auto timestamp = ocpp::DateTime(transaction_started.timestamp);
-                const auto meter_start = transaction_started.energy_Wh_import;
+                const auto meter_value =
+                    get_meter_value(transaction_started.meter_value, ocpp::v201::ReadingContextEnum::Transaction_Begin);
                 const auto session_id = session_event.uuid;
                 const auto signed_meter_value = transaction_started.signed_meter_value;
                 const auto reservation_id = transaction_started.reservation_id;
-
-                // meter start
-                ocpp::v201::MeterValue meter_value;
-                ocpp::v201::SampledValue sampled_value;
-                sampled_value.context = ocpp::v201::ReadingContextEnum::Transaction_Begin;
-                sampled_value.location = ocpp::v201::LocationEnum::Outlet;
-                sampled_value.measurand = ocpp::v201::MeasurandEnum::Energy_Active_Import_Register;
-                sampled_value.value = meter_start;
-                meter_value.sampledValue.push_back(sampled_value);
-                meter_value.timestamp = timestamp;
+                const auto remote_start_id = transaction_started.id_tag.request_id;
 
                 ocpp::v201::IdToken id_token;
-                id_token.idToken = transaction_started.id_tag;
-                id_token.type = ocpp::v201::IdTokenEnum::ISO14443;
+                id_token.idToken = transaction_started.id_tag.id_token;
+                if (transaction_started.id_tag.id_token_type.has_value()) {
```

Comment:
I think, the naming here with `id_tag` and `id_token_type` is still broken. But this might be due to predifined OCCP types? At least for me, `transaction_started.id_tag.id_token_type` really reads unnaturally